### PR TITLE
Added Config Utility

### PR DESCRIPTION
**Pull request by barckcode**
- Global Config Utility
- Global Log Utility
- Postgres Module (Only Config feature)

### CODE REVIEW

File: config.yml
Comment (pmorillas) [inline config.yml:1]: Proposed configuration scheme for better flexibility:

```yaml
datasources:
  - type: postgres
    name: psql-prod
    host: psql-prod.example.com
    port: 5432
    username: postgres
    password: password
  - type: postgres
    name: psql-stage
    host: psql-stage.example.com
    port: 5432
    username: postgres
    password: password
  - type: mssql
    name: mssql-prod
    host: mssql-prod.example.com
    port: 1433
    username: backup
    password: password
  - type: file
    name: local
    host: bastion.helmcode.com
    path: /var/opt/backup
    compression: true
  - type: s3
    bucket-uri: s3://backup.example.com
    path: /

jobs:
  - type: backup
    name: backup sauron
    schedule: ...
    source:
      - sauron
    destination:
      - s3
  - type: backup
    name: daily backup rds postgres
    schedule: 0 3 * * *
    source:
      - name: psql-prod
        dbs: []
      - mssql-prod
    destination:
      - local
      - s3
  - type: restore
    name: Restore prod to stage
    schedule: 0 5 * * 1
    source:
      name: local
      path: psql-prod
    destination: psql-stage
```

Comment (pmorillas) [reply config.yml:1]: We could use this tool as a manager for cron jobs or manual execution at the same time. This way we don't have to cron the script and poll jobs

File: config.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+---
+# Backup system configuration file
+#
+datasources:
+  - host: localhost
```
Comment (pmorillas) [inline config.yml:5]: What about file sources?

File: config.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        # exclude_dbs:[] - databases list to exclude from backup
+        # if not configured, it will backup all databases
+        dbs: []
+      - host: example.com
+        exclude_dbs: []
+    destination:
+      # Options:
+      # dir: directory to store backups
+      # s3: s3 bucket to store backups
+      dir: /var/lib/backup
```
Comment (pmorillas) [inline config.yml:34]: Is it possible to add multiple destinations?

File: utils/log.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+error_log() {
```
Comment (pmorillas) [inline utils/log.sh:2]: Invert naming to keep function names consistent? "log_error()"

File: config.yml
Comment (barckcode) [reply config.yml:1]: Added changes to the configuration file

File: utils/log.sh (hunk 3 above)
Comment (barckcode) [reply utils/log.sh:2]: Since it is an error log and not a specific error, it makes more sense to call it "error_log"

File: config.yml (hunk 2 above)
Comment (barckcode) [reply config.yml]: Yes, as long as you indicate it in the configuration file. Example:

```YAML
destination:
  dir: /home
  s3_uri: s3://my-bucket
```

File: config.yml (hunk 1 above)
Comment (barckcode) [reply config.yml]: Good point, its not included in the config file yet. It has to be implemented

Comment (barckcode) [reply config.yml]: This will be done in future implementations to allow:
- backup local files
- backup files from remote servers

**Review by misscarapan — APPROVED** [review]

**Review by pmorillas — APPROVED** [review]

**Pull request merged**